src/DropAutomation.sol

```diff
@@ -0,0 +1,288 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IQuoter} from "@interfaces/IQuoter.sol";
+import {ISwapRouter} from "@interfaces/ISwapRouter.sol";
+
+interface IRewardsDistributorSafeModule {
+    function addRewards(uint256 amountToken1, uint256 amountToken2) external;
+}
+
+/**
+ * @title DropAutomation
+ * @notice Automates weekly reward drops by swapping rewards into MAMO via Aerodrome and
+ *         forwarding the resulting MAMO/cbBTC balances to the F-MAMO Safe rewards module.
+ */
+contract DropAutomation is Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @dev MAMO token
+    IERC20 public immutable MAMO_TOKEN;
+
+    /// @dev cbBTC token
+    IERC20 public immutable CBBTC_TOKEN;
+
+    /// @dev F-MAMO Safe multisig address receiving reward tokens
+    address public immutable F_MAMO_SAFE;
+
+    /// @dev Safe module responsible for staging reward distributions
+    IRewardsDistributorSafeModule public immutable SAFE_REWARDS_DISTRIBUTOR_MODULE;
+
+    /// @dev Aerodrome CL router used to swap earned rewards into MAMO
+    ISwapRouter private constant AERODROME_CL_ROUTER = ISwapRouter(0xBE6D8f0d05cC4be24d5167a3eF062215bE6D18a5);
+
+    /// @dev Aerodrome quoter used to fetch swap estimates for slippage protection
+    IQuoter private constant AERODROME_QUOTER = IQuoter(0x254cF9E1E6e233aa1AC962CB9B05b2cfeAaE15b0);
+
+    /// @notice Address authorized to trigger reward drops
```

Comment:
Please set this in the constructor instead. Reason: we can reuse the same script for different chains where the addresses vary.